modules/consul-cluster/variables.tf

```diff
@@ -122,6 +122,32 @@ variable "termination_policies" {
   default     = "Default"
 }
 
+variable "lifecycle_hooks" {
+  description = "The lifecycle hooks to create that are triggered by the launch event. This is a map where the keys are the name of the hook and the values are an object with the keys and values defined in the lifecycle_hook block of the aws_autoscaling_group resource.  Default is no launch hooks"
+  type        = map(any)
+  default     = {}
+
+#    example = {
+#      name = {
+#        default_result          = string : CONTINUE | ABANDON
+#        heartbeat_timeout       = int
+#        lifecycle_transition    = string : "autoscaling:EC2_INSTANCE_LAUNCHING" | "autoscaling:EC2_INSTANCE_TERMINATING"
+#        notification_metadata   = string
+#        notification_target_arn = string
+#        role_arn                = string
+#    }
+
+  validation {
+    condition = (length([for x in values(var.lifecycle_hooks): x if !contains(["CONTINUE","ABANDON"], lookup(x,"default_result","CONTINUE"))])==0)
```

Comment:
NIT: could use [`anytrue`](https://www.terraform.io/docs/language/functions/anytrue.html).

NIT: please run `terraform fmt`. Spacing looks a bit off.

Reply:
done!  anytrue is much better.  sorry about the fmt, I thought it found nothing to change when I ran it but it seems like it didn't actually recurse when run in the root.